utility/src/main/java/org/oppia/util/parser/HtmlParser.kt

```diff
@@ -47,21 +45,17 @@ class HtmlParser private constructor(
 
     val imageGetter = urlImageParserFactory.create(htmlContentTextView, entityType, entityId, imageCenterAlign)
 
-    val htmlSpannable = HtmlCompat.fromHtml(htmlContent, HtmlCompat.FROM_HTML_MODE_LEGACY, imageGetter, LiTagHandler()) as Spannable
+    val formattedHtml = htmlContent
+      .replace("(?i)<ul[^>]*>".toRegex(), "<${StringUtils.UL_TAG}>")
+      .replace("(?i)</ul>".toRegex(), "</${StringUtils.UL_TAG}>")
+      .replace("(?i)<ol[^>]*>".toRegex(), "<${StringUtils.OL_TAG}>")
+      .replace("(?i)</ol>".toRegex(), "</${StringUtils.OL_TAG}>")
+      .replace("(?i)<li[^>]*>".toRegex(), "<${StringUtils.LI_TAG}>")
+      .replace("(?i)</li>".toRegex(), "</${StringUtils.LI_TAG}>")
+
+    val htmlSpannable = HtmlCompat.fromHtml(formattedHtml, HtmlCompat.FROM_HTML_MODE_LEGACY, imageGetter, LiTagHandler(htmlContentTextView.context)) as Spannable
```

Comment:
This should be in multiple lines as it crosses the 80 character limit.

Reply:
Done.